R/plot-individual-time-profile.R

```diff
@@ -0,0 +1,118 @@
+
+  # data frames -----------------------------
+
+  df <- dataCombined$toDataFrame()
+
+  # Getting all units on the same scale
+  df <- .unitConverter(df, defaultPlotConfiguration$xUnit, defaultPlotConfiguration$yUnit)
+
+  # Datasets which haven't been assigned to any group will be plotted as a group
+  # on its own. That is, the `group` column entries for them will be their names.
+  df <- .addMissingGroupings(df)
+
+  # `TimeProfilePlotConfiguration` object -----------------------------
+
+  # Create an instance of `TimeProfilePlotConfiguration` class by doing a
+  # one-to-one mapping of internal plot configuration object's public fields
+  populationTimeProfilePlotConfiguration <- .convertGeneralToSpecificPlotConfiguration(
+    data = df,
+    specificPlotConfiguration = tlf::TimeProfilePlotConfiguration$new(),
+    generalPlotConfiguration = defaultPlotConfiguration
+  )
+
+  # plot -----------------------------
+
+  # Which dataset types are present?
+  datasetTypePresent <- .extractPresentDatasetTypes(dataCombined)
+
+  # both observed and simulated
+  if (datasetTypePresent == .presentDataTypes$Both) {
+    obsData <- dplyr::filter(df, dataType == "observed")
+    simData <- dplyr::filter(df, dataType == "simulated")
+
+    profilePlot <- tlf::plotTimeProfile(
+      data = as.data.frame(simData),
+      dataMapping = tlf::TimeProfileDataMapping$new(
+        x = "xValues",
+        y = "yValues",
+        group = "group"
+      ),
+      observedData = as.data.frame(obsData),
+      observedDataMapping = tlf::ObservedDataMapping$new(
+        x = "xValues",
+        y = "yValues",
+        group = "group",
+        error = "yErrorValues"
+      ),
+      plotConfiguration = populationTimeProfilePlotConfiguration
+    )
+
+    # Extract color and shape mappings
+    legendCaptionData <- tlf::getLegendCaption(profilePlot)
+
+    # Extract as many colors as there are datasets from the specified color palette.
+    colorPalette <- defaultPlotConfiguration$pointsColor[1:nrow(legendCaptionData)]
+
+    # New version of legend mappings.
+    newLegendCaptionData <- dplyr::mutate(legendCaptionData, color = colorPalette)
+
+    # Update plot with these colors.
+    profilePlot <- tlf::updateTimeProfileLegend(profilePlot, caption = newLegendCaptionData)
+  }
+
+  # only observed
+  if (datasetTypePresent == .presentDataTypes$Observed) {
+    obsData <- dplyr::filter(df, dataType == "observed")
+
+    profilePlot <- tlf::plotTimeProfile(
+      observedData = as.data.frame(obsData),
+      observedDataMapping = tlf::ObservedDataMapping$new(
+        x = "xValues",
+        y = "yValues",
+        group = "group",
+        error = "yErrorValues"
+      ),
+      plotConfiguration = populationTimeProfilePlotConfiguration
+    )
+  }
+
+  # only simulated
+  if (datasetTypePresent == .presentDataTypes$Simulated) {
+    simData <- dplyr::filter(df, dataType == "simulated")
```

Comment:
Don't really have to filter here, no? "simulated" is the only data that are present, so you can just use "df"

Reply:
This code has been completely refactored after your suggestion, so this is outdated.